pkg/client/mount.go

```diff
@@ -224,18 +226,28 @@ func readlink(ctxt context.Context, device string) (string, error) {
 }
 
 func deviceFromMount(ctxt context.Context, file string) (string, error) {
-	cmd := []string{"sh", "-c", fmt.Sprintf(`"grep '%s' /proc/mounts | awk '{print \$1}'"`, file)}
-	out, err := co.RunCmd(ctxt, cmd...)
-	if err != nil {
-		return "", err
-	}
-	dev, err := readlink(ctxt, out)
+
+        // Multiple Unix pipes does not work well in Golang
+        // Breaking into 2 distinct commands
+        cmd1 :=  []string{"sh", "-c", fmt.Sprintf("grep '%s' /proc/mounts", file)}
+        out1, err1 := co.RunCmd(ctxt, cmd1...)
+
+        cmd2 :=  []string{"sh", "-c", fmt.Sprintf("echo '%s' | awk '{print $1}'", out1)}
```

Comment:
You can't use pipe here, you will have the same issue as before.
I would suggest doing this parsing in go code, e,g

package main

import "fmt"
import "strings"

func main() {
	mountString := "/dev/sda1 /opt/stack/cinder/cinder/volume/drivers/datera ext4 rw,relatime,errors=remount-ro 0 0"
	components := strings.Fields(mountString)
	fmt.Println(components[0])
}

Reply:
Refactored code using above example.